src/OpenLoco/Viewport.cpp

```diff
@@ -60,29 +61,50 @@ namespace OpenLoco::Ui
     SavedViewSimple Viewport::toSavedView() const
     {
         SavedViewSimple result;
-        result.mapX = view_x + (view_width >> 1);
-        result.mapY = view_y + (view_height >> 1);
+        const auto centre = getCentre();
+        result.viewX = centre.x;
+        result.viewY = centre.y;
         result.zoomLevel = static_cast<ZoomLevel>(zoom);
         result.rotation = getRotation();
         return result;
     }
 
+    viewport_pos Viewport::getCentre() const
+    {
+        return viewport_pos(view_x + view_width / 2, view_y + view_height / 2);
+    }
+
+    xy32 Viewport::getUiCentre() const
```

Comment:
Didn't we deprecate the xy32 type? Any particular reason for using it now?

Reply:
I want to join xy32 and point_t using the point_t name but xy32 implementation. The reason i want the xy32 implementation is it has access to the math vector functions so I've been tending to use xy32 as I've often needed to use one of those functions. In this particular case it doesn't make a difference though.

Reply:
Ah, right. I suppose we should consolidate/rename the types once we've made a dent in the outstanding PRs then.